audio.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+import simpleaudio as sa
+import scales
+from typing import Dict, List
+import keyword
+
+ALL_NOTES = scales.ALL_NOTES
+EVENT_DURATION = 0.25  # [seconds]: the time each note is played.
+SAMPLE_RATE = 44100
+LOWEST_A = 220  # [Hz]: frequency of A3, which is chosen to be the reference frequency, meaning it is the lowest note to be played.
+
+# Calculate frequencies and store them in two dicts for two intervals. This is
+# needed to play scales that smoothly increases or decreases.
+
+FREQUENCY_DICT = {}
+FREQUENCY_DICT_NEXT_OCTAVE = {}
+for note in ALL_NOTES:
```

Comment:
Generally, try to keep logic out of the module body as it is hard to test and debug (this is run on import).

This case is borderline. Don't change it now, just keep this in mind.